src/main/java/com/tterrag/registrate/AbstractRegistrate.java

```diff
@@ -123,6 +124,79 @@ void addRegisterCallback(NonNullConsumer<? super T> callback) {
         }
     }
 
+    public static final class CreativeModeTabModifier implements CreativeModeTab.Output {
```

Comment:
I think there is enough here to extract this to its own class, and have the Registration class as package-private in the same file

Reply:
I have split the modifier class off onto its own class, but unsure as to how would make the registration class package-private in the modifier class, as AbstractRegistrate needs visibility to it for the backend list field to keep track of registrations (`AbstractRegistrate#creativeModeTabsRegistrars`)